gpt_index/indices/transform_retriever.py

```diff
@@ -0,0 +1,23 @@
+from typing import List, Optional
```

Comment:
this import structure is a bit weird `from gpt_index.indices.transform_retriever` and may break for users later on if we move this flie somewhere else. 

Does it make sense to define a `gpt_index.retrievers` folder where we can put all retriever imports? (even if the files don't live in that folder, just imagine an __init__.py importing from all existing files)

Reply:
yep